src/externalized/ShippingDestinationModel.cc

```diff
@@ -0,0 +1,70 @@
+#include "ShippingDestinationModel.h"
+
+ShippingDestinationModel::ShippingDestinationModel(uint8_t locationId_,
+	uint16_t chargeRateOverNight_, uint16_t chargeRate2Days_, uint16_t chargeRateStandard_,
+	uint8_t flowersNextDayDeliveryCost_, uint8_t flowersWhenItGetsThereCost_,
+	bool canDeliver_, bool isPrimary_,
+	uint8_t deliverySectorId_, uint8_t deliverySectorZ_, int16_t deliverySectorGridNo_,
+	int16_t emailOffset_, int16_t emailLength_):
+		locationId(locationId_), 
+		chargeRateOverNight(chargeRateOverNight_), chargeRate2Days(chargeRate2Days_), chargeRateStandard(chargeRateStandard_),
+		flowersNextDayDeliveryCost(flowersNextDayDeliveryCost_), flowersWhenItGetsThereCost(flowersWhenItGetsThereCost_),
+		canDeliver(canDeliver_), isPrimary(isPrimary_),
+		deliverySectorId(deliverySectorId_), deliverySectorX(SECTORX(deliverySectorId_)), deliverySectorY(SECTORY(deliverySectorId_)),
+		deliverySectorZ(deliverySectorZ_), deliverySectorGridNo(deliverySectorGridNo_),
+		emailOffset(emailOffset_), emailLength(emailLength_) {}
+
+uint8_t ShippingDestinationModel::getDeliverySector() const
+{
+	return SECTOR(deliverySectorX, deliverySectorY);
+}
+
+ShippingDestinationModel* ShippingDestinationModel::deserialize(JsonObjectReader& obj)
+{
+	uint8_t destSectorId = 0, destSectorZ = 0;
+	int16_t destGridNo = 0;
+	bool isPrimary = false;
+	bool canDeliver = obj.getOptionalBool("canDeliver");
+	if (canDeliver) {
+		destSectorId = SECTOR_FROM_SECTOR_SHORT_STRING(obj.GetString("deliverySector"));
+		destSectorZ = obj.GetInt("deliverySectorZ");
+		destGridNo = obj.GetInt("deliverySectorGridNo");
+		isPrimary = obj.getOptionalBool("isPrimary");
+	}
+
+	return new ShippingDestinationModel(
+		obj.GetInt("locationId"),
+		obj.GetInt("chargeRateOverNight"), obj.GetInt("chargeRate2Days"), obj.GetInt("chargeRateStandard"),
+		obj.getOptionalInt("flowersNextDayDeliveryCost"), obj.getOptionalInt("flowersWhenItGetsThereCost"),
+		canDeliver, isPrimary,
+		destSectorId, destSectorZ, destGridNo,
```

Comment:
destSectorX and Y are not important?

Reply:
`destSectorId`== `SECTOR(destSectorX, destSectorY)`. So we only need to store either one of these 2  forms and derive the other..

In game this is used to locate the 2 airports.

Reply:
Yeah, but we use the individual coordinates in several locations directly ... besides it just being normal to initialize everything properly.

Reply:
Updated. Now we only initialize and store X and Y in the class.